xgboost.py

```diff
@@ -0,0 +1,59 @@
+import sys
+sys.path.append("..")
+
+import pandas as pd
+import numpy as np
+from xgboost import XGBClassifier
+from sklearn.model_selection import StratifiedKFold,cross_val_score
+from sklearn.metrics import f1_score,classification_report,confusion_matrix
+
+from data_preprocessing.preprocessor import Preprocessor
+
+class FraudDetector:
+    def __init__(self,df):
+        self.df = df.copy()
+        self.X_nan = None
+        self.y_nan = None
+        self.df_dropped = None
+        self.X,self.y = None,None
+        self.X_train,self.X_test,self.y_train,self.y_test = None,None,None,None
+        self.xgb = None
+        self.skf = StratifiedKFold(n_splits=5,random_state=42,shuffle=True)
+        
+    def preprocess(self):
+        preprocessor = Preprocessor(self.df)
+        preprocessor.clean()
+        
+    def extract_nan_features(self):
+        self.X_nan = self.df[self.df.isnull().T.any()].drop('FLAG',axis=1)
+        self.y_nan = self.df[self.df.isnull().T.any()]['FLAG']
+        
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `extract_nan_features` method, you are extracting rows with any NaN values. However, this might not be the best approach as it could lead to loss of important data. Instead, consider imputing missing values based on some strategy (mean, median, mode, etc.) or dropping only those rows where a certain percentage of data is missing.



<!-- This is an auto-generated comment by OSS CodeRabbit -->